cmd/thanos/bucket.go

```diff
@@ -135,9 +141,9 @@ func registerBucketVerify(m map[string]setupFunc, root *kingpin.CmdClause, name
 		}
 
 		if *repair {
-			v = verifier.NewWithRepair(logger, bkt, backupBkt, fetcher, issues)
+			v = verifier.NewWithRepair(logger, reg, bkt, backupBkt, fetcher, time.Duration(*deleteDelay), issues)
 		} else {
-			v = verifier.New(logger, bkt, fetcher, issues)
+			v = verifier.New(logger, reg, bkt, fetcher, time.Duration(*deleteDelay), issues)
```

Comment:
It would be super nice if we can move out of duration as late as possible, otherwise the unit is no typed.

Reply:
I didn't understand this. Verifier requires time.Duration and we are sending it time.Duration and in this function only we are getting deleteDelay.

Reply:
Request you to please suggest changes here, about what exactly I can change. Thanks in advance!

Reply:
We talked about it offline during out meeting. It would be nice to operate on `time.Duration` not on int64 if we don't have to.

Reply:
Both `Verifier.New` and `Verifier.NewWithRepair` take in time.Duration as a parameter. I have seen that. I didn't get where we are using int64 actually.

deleteDelay is of type `model.Duration` and I see the same approach taken everywhere in the code that we get the value, convert it into `model.Duration` and then use time.Duration(*deleteDelay) to get time.Duration. Where does int64 come into the picture here?

It is possible that I am missing something, that is the reason why I asked for a suggestion. Thanks in advance!

Reply:
No, you are right, I am stupid here sorry, thought we are casting to int, we are casting to `time.Duration` from `model.Duration` ... all good! My bad :man_facepalming:

Reply:
resolving this discussion thread now.